pyplatypus/segmentation/models/stacked_ensemble.py

```diff
@@ -0,0 +1,191 @@
+            dropout_layer = SpatialDropout2D(rate=self.dropout)
+        else:
+            dropout_layer = Dropout(rate=self.dropout)
+        return dropout_layer
+
+    def convolutional_layer(
+            self, filters: int, kernel_size: Tuple[int, int], activation: Optional[str] = "linear"
+    ) -> Union[SeparableConv2D, Conv2D]:
+        """
+        Returns the convolutional layer of the demanded type.
+
+        Parameters
+        ----------
+        filters: int
+            Integer, the dimensionality of the output space (i.e. the number of output filters in the convolution).
+        kernel_size: Tuple[int, int])
+            An integer or tuple of 2 integers, specifying the width and height of the 2D convolution window.
+            If single integer is supplied the same value is used for all spatial dimensions.
+        activation: Optional[str]
+            Activation function name.
+
+        Returns
+        -------
+        convolutional layer.
+        """
+        if self.use_separable_conv2d:
+            convolutional_layer = SeparableConv2D(
+                filters=filters, kernel_size=kernel_size, padding="same",
+                kernel_initializer=self.kernel_initializer, activation=activation)
+        else:
+            convolutional_layer = Conv2D(
+                filters=filters, kernel_size=kernel_size, padding="same",
+                kernel_initializer=self.kernel_initializer, activation=activation)
+        return convolutional_layer
+
+    def activation(self):
+        """Creates the layer applying the specified activation function.
+
+        Returns
+        -------
+        activation_layer: function
+            Layer later used to apply the chosen activation function.
+        """
+        activation_layer = getattr(KRACT, self.activation_layer)
+        return activation_layer
+
+    def multiple_conv2d(
+            self,
+            input: tf.Tensor
+    ) -> tf.Tensor:
+        """
+        Creates a multiple convolutional U-Net block.
+
+        Parameters
+        ----------
+        input: tf.Tensor
+            Model or layer object.
+
+        Returns
+        -------
+        input:
+            Multiple convolutional block of the model.
+        """
+        for i in range(self.u_net_conv_block_width):
+            input = self.convolutional_layer(self.filters, self.kernel_size)(input)
+            if self.batch_normalization:
+                input = BatchNormalization()(input)
+            input = self.activation()(input)
+        return input
+
+    def build_model(self) -> tf.keras.Model:
+        """
+        Creates a semantic segmentation stacked ensemble architecture.
+
+        Returns
+        -------
+        model:
+            Semantic segmentation stacked ensemble model.
+        """
+        for i in range(len(self.submodels)):
+            submodel = self.submodels[i]
+            if self.copy_submodels_weights and self.freeze_submodels_weights:
```

Comment:
What do we want to do in case of copy_submodels_weights = False but freeze_submodels_weights = True?